src/terminal/parser/stateMachine.cpp

```diff
@@ -1634,21 +1656,43 @@ void StateMachine::_EventDcsPassThrough(const wchar_t wch)
 }
 
 // Routine Description:
-// - Handle the two-character termination of a DCS sequence.
+// - Handle SOS/PM/APC string
 //   Events in this state will:
-//   1. Enter ground on a string terminator
-//   2. Pass on everything else as the start of a regular escape sequence
+//   1. If we see a ESC, enter the SosPmApcTermination state.
+//   2. Ignore everything else.
 // Arguments:
 // - wch - Character that triggered the event
 // Return Value:
 // - <none>
-void StateMachine::_EventDcsTermination(const wchar_t wch)
+void StateMachine::_EventSosPmApcString(const wchar_t wch) noexcept
 {
-    _trace.TraceOnEvent(L"DcsTermination");
+    _trace.TraceOnEvent(L"SosPmApcString");
+    if (_isEscape(wch))
+    {
+        _EnterSosPmApcTermination();
+    }
+    else
+    {
+        _ActionIgnore();
+    }
+}
 
+void StateMachine::_EventVariableLengthStringTermination(const wchar_t wch)

```

Comment:
This ought to fix the ESC termination issue described here https://github.com/microsoft/terminal/pull/6328/files#r459167887